---
title: "Fig1e-Phyloseq"
author: "Linnea Honeker"
date: "1/18/2024"
output: html_document
---
# mac analysis for Parker

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library("phyloseq")
library(knitr)
library(apeglm)
library("ggplot2")
library("pheatmap")
library("stringr")
library(tidyverse)
library(reshape2)
library(heatmaply)
library(dada2)
library(phangorn)
library(microbiome)
library(RColorBrewer)
library(pgirmess)
library(vegan)
library(factoextra)
library(ggfortify)
library(ggnewscale)
library(ggrepel)


col_list <- c( "chartreuse3", "darkseagreen1","orange","lightgoldenrod","blue","lightblue", 
                             "orange4", "yellow2", "red4","tan", "brown1","darkorchid", "plum2","darkturquoise", "aquamarine2",
                              "hotpink4", "lightpink", "slateblue4", "lightslateblue", "darkgoldenrod", "moccasin","navy", "lightsteelblue3", 
                       "darksalmon","bisque", "blueviolet", "violet","cyan4","cyan",
                             "coral4","coral","darkred", "deeppink",
                             "darkslateblue","cornflowerblue")


# parameters for plots
h = 6
w = 7
res = 300
size = 10
```

## Create phyloseq object
```{r phyloseq object, echo=FALSE}
###load data and reformat to convert to phyloseq object
#look at microbial community composition at time 0
data.all <- read.table("../data/ASV_table_reads_with_sample_info.txt", header=T, sep="\t", stringsAsFactors=T, check.names=F) #Note that this input has total copies of 16S in original binned 
#SIP fractions to account for the different volumes - See line 435 for further explanation

#reanme treatment column for downstream analyses
data.all.2 <- data.all %>%
    mutate(treatment = gsub('50%', 'm50', treatment)) %>%
    mutate(treatment = gsub('100%', 'm100', treatment))


##load data and reformat to convert to phyloseq object -> normalize read counts to copy numbers per fraction
# First, create a dataframe showing fraction of tube total copies in each fraction
copies_norm <- data.all.2 %>%
  select( Replicate, treatment, Isotope,
          Hour, fraction, copies.ul.total) %>%
    unite(SampleID, c("treatment", "Hour", "Replicate", "Isotope")) %>%
  spread(key = fraction, value = copies.ul.total) %>%
  mutate(copies.tube = f1 + f2 + f3 + f4 + f5) %>%
  mutate(f1_norm = f1/copies.tube) %>%
  mutate(f2_norm = f2/copies.tube) %>%
  mutate(f3_norm = f3/copies.tube) %>%
  mutate(f4_norm = f4/copies.tube) %>%
  mutate(f5_norm = f5/copies.tube) %>%
  select(SampleID, f1_norm, f2_norm, f3_norm, f4_norm, f5_norm)

#create data frame showing relative abundance of each ASV
#gather dataframe to make calculated relative abundances easier

counts_RA.0 <- data.all.2 %>%
 select(Replicate, treatment, 
          Hour, Isotope, fraction, 
          where( ~ is.numeric(.x) && sum(.x) != 0), -tube, -Density, -copies.ul.total, -copies.ul) %>%
    unite(SampleID, c("treatment", "Hour", "Replicate", "Isotope")) %>%
  gather(key = "ASV_ID", value = "Counts", -c(SampleID, fraction)) %>%
  group_by(SampleID, fraction) %>%
  mutate(RelAbun = Counts / sum(Counts))

#spread dataframe of RAs back out
counts_RA <- counts_RA.0 %>%
  select(-Counts) %>%
  spread(key = ASV_ID, value = RelAbun)

#add a total reads column 
counts_total <- data.all.2 %>%
 select(Replicate, treatment, 
          Hour, Isotope, fraction, 
          where( ~ is.numeric(.x) && sum(.x) != 0), -tube, -Density, -copies.ul.total, -copies.ul) %>%
    unite(SampleID, c("treatment", "Hour", "Replicate", "Isotope")) %>%
  mutate(total_reads = rowSums(.[3:4263]))

#create column with new count (old total count per fraction * normalized copy for each fraction). First, merge total coutns with copies_norm
reads_new_total <- copies_norm %>%
  merge(counts_total, by = "SampleID") %>%
  select(SampleID, f1_norm, f2_norm, f3_norm, f4_norm, f5_norm, total_reads) %>%
  mutate(f1 = as.integer(f1_norm * total_reads)) %>%
  mutate(f2 = as.integer(f2_norm * total_reads)) %>%
  mutate(f3 = as.integer(f3_norm * total_reads)) %>%
  mutate(f4 = as.integer(f4_norm * total_reads)) %>%
  mutate(f5 = as.integer(f5_norm * total_reads)) %>%
  select(-f1_norm, -f2_norm, -f3_norm, -f4_norm, -f5_norm, -total_reads) %>%
  gather(key = "fraction", value = "new_count", -SampleID)

#merge new total counts with relative abundance table. Multiply each ASV rel abundance by new total count. Summarize by counts per tube.
norm_ASV_table <- reads_new_total %>%
  merge(counts_RA, by = c("SampleID", "fraction")) %>%
   mutate_each(funs(.*new_count), starts_with("sip")) %>%
  separate(SampleID, c("treatment", "Hour", "Replicate", "Isotope"), remove = TRUE) %>%
  select(-new_count) %>%
  group_by(treatment, Hour, Isotope,  Replicate) %>%
  summarise_at(vars(sip1:sip9983), ~ sum (.x, na.rm = TRUE)) %>%
  mutate_if(is.numeric, as.integer) %>%
  unite(SampleID, c("treatment", "Hour", "Replicate", "Isotope"), remove = FALSE) %>%
  column_to_rownames(var = "SampleID")

#transpose ASV table so that samples are rows and asvs are columns

#norm_ASV_tabel.t <- t(norm_ASV_table)

#separate out metadata and otu table matrix
data.otu <- norm_ASV_table %>%
  select(sip1:sip9983) 

data.otu.t <- as.matrix(t(data.otu))


#convert replicate to assocaited plot it came from
data.meta <- norm_ASV_table %>%
 select(treatment, Hour, Replicate, Isotope) %>%
  mutate(subject_id = case_when(
    (Replicate == "rA" & treatment == "m50") ~ "P09",
     (Replicate == "rB" & treatment == "m50") ~ "P10",
     (Replicate =="rC" & treatment == "m50") ~ "P13",
     (Replicate == "rA" & treatment == "m100") ~ "P08",
     (Replicate == "rB" & treatment == "m100") ~ "P14",
     (Replicate =="rC" & treatment == "m100") ~ "P16")
    )


#Import taxa legend:
taxa.id <- read.table("../data/taxonomy_table_reads_USEME.txt", 
                      header=F, sep="\t", stringsAsFactors=T, check.names=F, 
                      na.strings = "", col.names = c("taxon","code"))
taxa.id2 <- taxa.id %>%
  column_to_rownames(var = "taxon") %>%
  separate(code, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"), sep = ";")

taxa.mat <- as.matrix(taxa.id2)

  

####create phyloseq objects

otu = otu_table(data.otu.t, taxa_are_rows = TRUE)

map = sample_data(data.meta)

tax = tax_table(taxa.mat,errorIfNULL = TRUE)


#merge otu-table, sample-metadata, taxonomy, and tree
Data= merge_phyloseq(otu,map,tax)#,tree)
Data
taxa_are_rows(Data)



#filter 50% moisture
Data.50 = subset_samples(Data, treatment == "m50")

#filter to 100% moisture
Data.100 = subset_samples(Data, treatment == "m100")

#filter to 16O
Data.16O = subset_samples(Data, Isotope == "16O")




```

```{r create 'omics dataset}
#create omics dataset with universal sample ids to be combined with other omics
#transpose and take log2  
otu.log.0 <- log2(data.otu + 1)

otu.omics <- as.data.frame(otu.log.0) %>%
  rownames_to_column(var = "SampleID") %>%
  separate(SampleID, c("moisture", "timepoint", "Replicate", "isotope"), sep = "_") %>%
  mutate(moisture = gsub("m50", "x50", moisture)) |>
  mutate(moisture = gsub("m100", "x100", moisture)) |>
   mutate(subject_id = case_when(
    (Replicate == "rA" & moisture == "x50") ~ "P09",
     (Replicate == "rB" & moisture == "x50") ~ "P10",
     (Replicate =="rC" & moisture == "x50") ~ "P13",
     (Replicate == "rA" & moisture == "x100") ~ "P08",
     (Replicate == "rB" & moisture == "x100") ~ "P14",
     (Replicate =="rC" & moisture == "x100") ~ "P16")
    ) |>
  filter(isotope == "16O") |>
  unite(SampleID, c("subject_id", "moisture", "timepoint"), sep = "_") |>
  select(-isotope, -Replicate) |>
  column_to_rownames(var = "SampleID")

write.csv(otu.omics, "../output/omics_datasets/otu.csv")

#save taxonomy table
taxa.omics <- as.data.frame(taxa.mat)
write.csv(taxa.omics, "../output/omics_datasets/taxa.csv")

#save metadata table
meta.omics <- data.meta %>%
  rownames_to_column(var = "SampleID") %>%
  separate(SampleID, c("moisture", "timepoint", "Replicate", "isotope"), sep = "_") %>%
 mutate(moisture = gsub("m50", "x50", moisture)) %>%
  mutate(moisture = gsub("m100", "x100", moisture)) %>%
   mutate(subject_id = case_when(
    (Replicate == "rA" & moisture == "x50") ~ "P9",
     (Replicate == "rB" & moisture == "x50") ~ "P10",
     (Replicate =="rC" & moisture == "x50") ~ "P13",
     (Replicate == "rA" & moisture == "x100") ~ "P8",
     (Replicate == "rB" & moisture == "x100") ~ "P14",
     (Replicate =="rC" & moisture == "x100") ~ "P16")
    ) |>
  filter(isotope == "16O") %>%
  unite(SampleID, c("subject_id", "moisture", "timepoint"), sep = "_", remove = FALSE) %>%
  select(-isotope, -Replicate, -Hour, -Isotope, -Replicate, -treatment) %>%
  arrange(SampleID)%>%
  column_to_rownames(var = "SampleID")

write.csv(meta.omics, "../output/omics_datasets/metadata.csv")

```


## Steps:
### filter phylseq object to remove low sequence counts ( < 0.01%) and rarefy
```{r filter, include  = FALSE}
# all depths
total_counts <-sum(taxa_sums(Data))
threshold <- 0.00001*total_counts
Data.f <- filter_taxa(Data, function(x) sum(x) > threshold, TRUE)

Data.f.ra = rarefy_even_depth(Data.f,(min(sample_sums(Data.f))))

# low moisture
total_counts <-sum(taxa_sums(Data.50))
threshold <- 0.00001*total_counts
Data.50.f <- filter_taxa(Data.50, function(x) sum(x) > threshold, TRUE)

Data.50.f.ra = rarefy_even_depth(Data.50.f,(min(sample_sums(Data.50.f))))

# high moisture
total_counts <-sum(taxa_sums(Data.100))
threshold <- 0.00001*total_counts
Data.100.f <- filter_taxa(Data.100, function(x) sum(x) > threshold, TRUE)

Data.100.f.ra = rarefy_even_depth(Data.100.f,(min(sample_sums(Data.100.f))))

# 16O
total_counts <-sum(taxa_sums(Data.16O))
threshold <- 0.00001*total_counts
Data.16O.f <- filter_taxa(Data.16O, function(x) sum(x) > threshold, TRUE)

Data.16O.f.ra = rarefy_even_depth(Data.16O.f,(min(sample_sums(Data.16O.f))))
```

### taxonomy plots
phylum 16O only
```{r taxonomy phylum, echo = FALSE}

#taxonomy--------------------------------------------------------
#prune out phyla below 2% in each sample for all experiments
Data.f.rel.abun <- transform_sample_counts(Data.16O.f, function(x) x/sum(x)) #convert counts to relative abundances
Data.f.glom <- tax_glom(Data.f.rel.abun, taxrank = "Phylum") #agglomerate taxa to Phylum-level
Data.f.dat <- psmelt(Data.f.glom)
Data.f.dat$Phylum <- as.character(Data.f.dat$Phylum) #convert to character vector from factor
Data.f.dat$Phylum[Data.f.dat$Abundance < 0.02] <- "Phylum < 2%"



#######plot Phylum######################################
#all time points
#change order of facets (facet by moisture)
Data.f.dat$Hour <- factor(Data.f.dat$Hour, 
                        levels =c("t0", "t3", "t24", "t48", "t72", "t168"))
Data.f.dat$treatment <- factor(Data.f.dat$treatment, 
       levels =c("m50", "m100")) 

Data.f.dat <- Data.f.dat %>%
  mutate(Phylum = str_remove(Data.f.dat$Phylum, "p_"))

p <- ggplot(Data.f.dat, aes(x=Hour, y = Abundance, fill=Phylum)) 

p.phylum <- p + geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values=col_list_2) +
  facet_wrap(~ treatment) +
  labs(y= "Relative abundance") +
  scale_x_discrete(labels = c("0", "3", "24", "48", "72", "168")) +
  theme_bw() +
  theme(text = element_text(size = 16),
        axis.text.x=element_text(size=12, angle=0,hjust =0.5)) 

p.phylum
ggsave("../figures/Fig1-taxon-specific-growth-rates/bulk-phylum-timepoint-facet-moisture.png",units=c('in'),width=8,height=5,dpi=300,p.phylum)
ggsave("../figures/Fig1-taxon-specific-growth-rates/bulk-phylum-timepoint-facet-moisture.pdf",units=c('in'),width=8,height=5,dpi=300,p.phylum)


```


```{r lme phyla}
##Linear mixed effects models to determine significant differences in phyla across moisture treatments
##Proteobacteria
Data.f.dat.Proteo <- Data.f.dat %>%
  filter(Phylum == "Proteobacteria")

m <- lmer(Abundance ~ treatment * Hour + (1|subject_id), data = Data.f.dat.Proteo)
anova(m)

# Type III Analysis of Variance Table with Satterthwaite's method
#                  Sum Sq  Mean Sq NumDF DenDF F value    Pr(>F)    
# treatment      0.082699 0.082699     1     4 80.9187 0.0008455 ***
# Hour           0.061506 0.012301     5    20 12.0364 1.799e-05 ***
# treatment:Hour 0.010319 0.002064     5    20  2.0193 0.1194987    

# Post-hoc pairwise contrasts at each timepoint
em    <- emmeans(m, ~ treatment | Hour)
pairs(em, adjust="tukey")

# Hour = t0:
#  contrast   estimate     SE   df t.ratio p.value
#  m50 - m100  -0.1455 0.0277 22.6  -5.250  <.0001
# 
# Hour = t3:
#  contrast   estimate     SE   df t.ratio p.value
#  m50 - m100  -0.1908 0.0277 22.6  -6.886  <.0001
# 
# Hour = t24:
#  contrast   estimate     SE   df t.ratio p.value
#  m50 - m100  -0.0993 0.0277 22.6  -3.582  0.0016
# 
# Hour = t48:
#  contrast   estimate     SE   df t.ratio p.value
#  m50 - m100  -0.1311 0.0277 22.6  -4.731  0.0001
# 
# Hour = t72:
#  contrast   estimate     SE   df t.ratio p.value
#  m50 - m100  -0.1026 0.0277 22.6  -3.704  0.0012
# 
# Hour = t168:
#  contrast   estimate     SE   df t.ratio p.value
#  m50 - m100  -0.0941 0.0277 22.6  -3.394  0.0025

#Firmicutes
Data.f.dat.Firm <- Data.f.dat %>%
  filter(Phylum == "Firmicutes")

m <- lmer(Abundance ~ treatment * Hour + (1|subject_id), data = Data.f.dat.Firm)
anova(m)

# Type III Analysis of Variance Table with Satterthwaite's method
#                   Sum Sq   Mean Sq NumDF DenDF F value    Pr(>F)    
# treatment      0.0000657 0.0000657     1     4  0.1690    0.7021    
# Hour           0.0219776 0.0043955     5    20 11.3004 2.817e-05 ***
# treatment:Hour 0.0005140 0.0001028     5    20  0.2643    0.9273

# Post-hoc pairwise contrasts at each timepoint
em    <- emmeans(m, ~ treatment | Hour)
pairs(em, adjust="tukey") #no sig diff

#Actinobacteria
Data.f.dat.Actino <- Data.f.dat %>%
  filter(Phylum == "Actinobacteria")

m <- lmer(Abundance ~ treatment * Hour + (1|subject_id), data = Data.f.dat.Actino)
anova(m)

# Type III Analysis of Variance Table with Satterthwaite's method
#                  Sum Sq  Mean Sq NumDF DenDF  F value    Pr(>F)    
# treatment      0.076532 0.076532     1    24 109.6149 1.991e-10 ***
# Hour           0.055067 0.011013     5    24  15.7742 6.555e-07 ***
# treatment:Hour 0.006349 0.001270     5    24   1.8188    0.1472  

# Post-hoc pairwise contrasts at each timepoint
em    <- emmeans(m, ~ treatment | Hour)
pairs(em, adjust="tukey") 

# Hour = t0:
#  contrast   estimate     SE df t.ratio p.value
#  m50 - m100   0.1216 0.0216 24   5.635  <.0001
# 
# Hour = t3:
#  contrast   estimate     SE df t.ratio p.value
#  m50 - m100   0.1265 0.0216 24   5.862  <.0001
# 
# Hour = t24:
#  contrast   estimate     SE df t.ratio p.value
#  m50 - m100   0.0745 0.0216 24   3.454  0.0021
# 
# Hour = t48:
#  contrast   estimate     SE df t.ratio p.value
#  m50 - m100   0.1038 0.0216 24   4.811  0.0001
# 
# Hour = t72:
#  contrast   estimate     SE df t.ratio p.value
#  m50 - m100   0.0704 0.0216 24   3.263  0.0033
# 
# Hour = t168:
#  contrast   estimate     SE df t.ratio p.value
#  m50 - m100   0.0565 0.0216 24   2.620  0.0150


#Bacteriodetes
Data.f.dat.Bact <- Data.f.dat %>%
  filter(Phylum == "Bacteroidetes")

m <- lmer(Abundance ~ treatment * Hour + (1|subject_id), data = Data.f.dat.Bact)
anova(m)

# Type III Analysis of Variance Table with Satterthwaite's method
#                    Sum Sq    Mean Sq NumDF DenDF F value    Pr(>F)    
# treatment      0.00185506 0.00185506     1    22 35.5655 5.304e-06 ***
# Hour           0.00104545 0.00020909     5    22  4.0087  0.009765 ** 
# treatment:Hour 0.00045017 0.00009003     5    22  1.7261  0.170467    

# Post-hoc pairwise contrasts at each timepoint
em    <- emmeans(m, ~ treatment | Hour)
pairs(em, adjust="tukey") 

# Hour = t0:
#  contrast   estimate     SE df t.ratio p.value
#  m50 - m100 -0.01378 0.0067 22  -2.058  0.0516
# 
# Hour = t3:
#  contrast   estimate     SE df t.ratio p.value
#  m50 - m100 -0.02749 0.0067 22  -4.105  0.0005
# 
# Hour = t24:
#  contrast   estimate     SE df t.ratio p.value
#  m50 - m100 -0.00611 0.0059 22  -1.035  0.3117
# 
# Hour = t48:
#  contrast   estimate     SE df t.ratio p.value
#  m50 - m100 -0.01961 0.0059 22  -3.326  0.0031
# 
# Hour = t72:
#  contrast   estimate     SE df t.ratio p.value
#  m50 - m100 -0.00643 0.0059 22  -1.090  0.2874
# 
# Hour = t168:
#  contrast   estimate     SE df t.ratio p.value
#  m50 - m100 -0.01624 0.0059 22  -2.754  0.0116


#Verrucomicrobia
Data.f.dat.Verr <- Data.f.dat %>%
  filter(Phylum == "Verrucomicrobia")

m <- lmer(Abundance ~ treatment * Hour + (1|subject_id), data = Data.f.dat.Verr)
anova(m)

# Type III Analysis of Variance Table with Satterthwaite's method
#                    Sum Sq    Mean Sq NumDF DenDF F value  Pr(>F)  
# treatment      7.5653e-05 7.5653e-05     1     4  4.5079 0.10098  
# Hour           2.3490e-04 4.6980e-05     5    20  2.7994 0.04487 *
# treatment:Hour 3.7579e-05 7.5160e-06     5    20  0.4478 0.80980    

# Post-hoc pairwise contrasts at each timepoint
em    <- emmeans(m, ~ treatment | Hour)
pairs(em, adjust="tukey") #no sig
```

